modules/opensearch/main.tf

```diff
@@ -0,0 +1,280 @@
+    enabled = true
+  }
+
+  advanced_security_options {
+    enabled = true
+    internal_user_database_enabled = true
+
+    master_user_options {
+      master_user_name     = var.master_user_name
+      master_user_password = var.master_user_password
+    }
+  }
+
+  node_to_node_encryption {
+    enabled = true
+  }
+
+  domain_endpoint_options {
+    custom_endpoint_certificate_arn = var.custom_endpoint_certificate_arn
+    custom_endpoint_enabled         = true
+    custom_endpoint                 = "${local.name}.${var.dns_base_domain}"
+    enforce_https                   = true
+    tls_security_policy             = "Policy-Min-TLS-1-2-2019-07"
+  }
+
+  cluster_config {
+    instance_type  = var.opensearch_instance_type
+    instance_count = var.opensearch_instance_count
+  }
+
+  ebs_options {
+    ebs_enabled = true
+    volume_type = var.opensearch_ebs_volume_type
+    volume_size = var.opensearch_ebs_volume_size
+  }
+
+  log_publishing_options {
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.opensearch.arn
+    log_type                 = "INDEX_SLOW_LOGS"
+  }
+
+  log_publishing_options {
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.opensearch.arn
+    log_type                 = "SEARCH_SLOW_LOGS"
+  }
+
+  log_publishing_options {
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.opensearch.arn
+    log_type                 = "ES_APPLICATION_LOGS"
+  }
+
+  log_publishing_options {
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.opensearch.arn
+    log_type                 = "AUDIT_LOGS"
+  }
+
+  tags = local.tags
+
+  depends_on = [aws_iam_service_linked_role.this]
+}
+
+# OpenSearch service-linked role
+resource "aws_iam_service_linked_role" "this" {
+  aws_service_name = "opensearchservice.amazonaws.com"
+}
+
+# Access policy
+resource "aws_opensearch_domain_policy" "this" {
+  domain_name = aws_opensearch_domain.this.domain_name
+
+  access_policies = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect    = "Allow"
+        Action    = "es:*"
+        Principal = "*"
+        Resource  = "${aws_opensearch_domain.this.arn}/*"
+      }
+    ]
+  })
+}
+
+# Logs
+resource "aws_cloudwatch_log_group" "opensearch" {
+  name = "/aws/OpenSearchService/domains/${local.name}"
+}
+
+resource "aws_cloudwatch_log_resource_policy" "this" {
+  policy_name = "OpenSearchService-${local.name}"
+
+  policy_document = <<CONFIG
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "es.amazonaws.com"
+      },
+      "Action": [
+        "logs:PutLogEvents",
+        "logs:PutLogEventsBatch",
+        "logs:CreateLogStream"
+      ],
+      "Resource": "arn:aws:logs:*"
+    }
+  ]
+}
+CONFIG
+}
+
+# CNAME for custom OpenSearch endpoint
+resource "aws_route53_record" "this" {
+  zone_id = var.hosted_zone_id
+  name    = "${local.name}.${var.dns_base_domain}"
+  type    = "CNAME"
+  ttl     = "5"
```

Comment:
How come such a low ttl? Maybe this could also be a variable?

Reply:
Changed to 600 and made it a variable